lib/crawly/pipelines/csv_encoder.ex

```diff
@@ -17,13 +17,8 @@ defmodule Crawly.Pipelines.CSVEncoder do
   @spec run(map, map, fields: list(atom)) ::
           {false, state :: map} | {csv_line :: String.t(), state :: map}
   def run(item, state, opts \\ []) do
-    opts = Enum.into(opts, %{fields: nil})
-    fields = Map.get(opts, :fields)
-
-    case fields do
-      nil ->
-        # only for when both tuple and global config is not provided
-
+    case opts[:fields] do
+      fields when fields in [nil, []] ->
```

Comment:
Hmm it doesn't work - [this](https://github.com/michaltrzcinka/crawly/blob/f63f80aaef8cb7c0d50cc9b30e5f0cfeea20ecf5/test/pipelines/csv_encoder_test.exs#L29) assertion is failing.
I think it's because `Map.get` will return a default value only if the map doesn't contain given key. But in our case the map has such key, its value just happens to be `nil`.